### PR TITLE
Add support for 3.1.0-alpha01

### PR DESCRIPTION
**Pull request by jaredsburrows**
Closes https://github.com/KeepSafe/dexcount-gradle-plugin/issues/224
@benjamin-bader 

We need https://github.com/KeepSafe/dexcount-gradle-plugin/pull/226 merged first

### CODE REVIEW

File: src/main/kotlin/com/getkeepsafe/dexcount/Tasks.kt [hunk 1]
```diff
@@ -19,11 +19,11 @@ package com.getkeepsafe.dexcount
 import com.android.build.gradle.api.BaseVariantOutput
 import org.gradle.api.DefaultTask
 import org.gradle.api.GradleException
-import org.gradle.api.Nullable
```
Comment (jaredsburrows) [inline src/main/kotlin/com/getkeepsafe/dexcount/Tasks.kt]: Deprecated

File: src/main/kotlin/com/getkeepsafe/dexcount/Plugin.kt [hunk 2]
```diff
@@ -40,22 +39,42 @@ import kotlin.reflect.KClass
...
+        if (gradlePluginVersion == null && exception != null) {
+            throw IllegalStateException("dexcount requires the Android plugin to be configured", exception)
+        } else if (gradlePluginVersion == null) {
+            throw IllegalStateException("dexcount requires the Android plugin to be configured")
         }
 
         sdkLocation = SdkResolver.resolve(project)
 
-        val gradlePluginRevision = Revision.parseRevision(Version.ANDROID_GRADLE_PLUGIN_VERSION)
+        val gradlePluginRevision = Revision.parseRevision(gradlePluginVersion, Revision.Precision.PREVIEW)
```
Comment (jaredsburrows) [inline src/main/kotlin/com/getkeepsafe/dexcount/Plugin.kt]: `Revision.Precision.PREVIEW` handles `x.y.z-preview`

**Comment by jaredsburrows** [comment]
@benjamin-bader Rebased!

**Comment by jaredsburrows** [comment]
@benjamin-bader It looks like the method count has changed while updating plugins? hmm. I am guessing this is because it is using this new dex compiler - https://androidstudio.googleblog.com/2017/10/android-studio-31-canary-1-now-available.html.

**Comment by benjamin-bader** [comment]
Sounds right; it tends to change along with tool versions

**Comment by jaredsburrows** [comment]
@benjamin-bader Should I update the assertions to match the new 3.1 tool?

**Comment by benjamin-bader** [comment]
Yeah, that's what I'd do.  TBH, I just update the test to match what dexcount produces when tool versions change; I trust that the dex file format is consistent, so I trust the output of dexcount - so long as the counting logic hasn't _also_ changed, which it hasn't here.

**Comment by jaredsburrows** [comment]
@benjamin-bader Ok. U updated the test assertions to work with the new plugin and I added missing ones as well.

**Comment by jaredsburrows** [comment]
@benjamin-bader Safe to merge?

**Comment by benjamin-bader** [comment]
Thanks, Jared!

**Pull request merged**